### PR TITLE
Re-load most recently ran script 

### PR DESCRIPTION
**Pull request by TamTrongNguyen**
Adds a shortcut to re-load the most recently ran script 

https://github.com/mgba-emu/mgba/issues/2845

### CODE REVIEW

File: src/platform/qt/Window.cpp [hunk 1]
```diff
@@ -1632,6 +1632,11 @@ void Window::setupMenu(QMenuBar* menubar) {
 	addGameAction(tr("&Cheats..."), "cheatsWindow", openControllerTView<CheatsView>(), "tools");
 #ifdef ENABLE_SCRIPTING
 	m_actions.addAction(tr("Scripting..."), "scripting", this, &Window::scriptingOpen, "tools");
+	
+	addGameAction(tr("Load recent script"), "loadscript", [this] {
+		m_mruFiles = m_config->getMRU(ConfigController::MRU::Script);
```
Comment (endrift) [inline src/platform/qt/Window.cpp:1637]: This overwrites the member variable `m_mruFiles`, which is _not_ what you want.

**Review by endrift — CHANGES_REQUESTED** [review]
This needs to appear in the scripting window's menu, not the parent window's menu.

File: src/platform/qt/Window.cpp (hunk 1 above)
Comment (TamTrongNguyen) [reply src/platform/qt/Window.cpp]: I changed the location of the shortcut to appear in the scripting window but was not able to bind it to any keys. It does not overwrite m_mruFiles and does play the most recently run script in the scripting window

File: src/platform/qt/scripting/ScriptingView.cpp [hunk 2]
```diff
@@ -62,10 +63,15 @@ void ScriptingView::load() {
 		if (!m_controller->loadFile(filename)) {
 			return;
 		}
+		setMRS(filename);
```
Comment (endrift) [inline src/platform/qt/scripting/ScriptingView.cpp:66]: You don't need to store this separately, it's always going to be `m_mruFiles.at(0)`--unless `m_mruFiles.isEmpty()` is `true`, in which case there is no most recently loaded script.

File: src/platform/qt/scripting/ScriptingView.h [hunk 3]
```diff
@@ -22,6 +22,7 @@ Q_OBJECT
 private slots:
 	void submitRepl();
 	void load();
+	void recent();
```
Comment (endrift) [inline src/platform/qt/scripting/ScriptingView.h:25]: I'd recommend the name `loadRecent` instead, as `recent` isn't a verb and lacks clarity of the action in question

**Review by endrift — CHANGES_REQUESTED** [review]

**Comment by endrift** [comment]
Are you planning to finish this PR now that the semester is (presumably) over?

**Pull request closed without merging**